src/database.ts

```diff
@@ -310,7 +310,32 @@ class Database extends common.GrpcServiceObject {
         options: CreateDatabaseOptions,
         callback: CreateDatabaseCallback
       ) => {
-        return instance.createDatabase(formattedName_, options, callback);
+        const pool = this.pool_ as SessionPool;
+        if (pool._pending > 0) {
+          // If there are BatchCreateSessions requests pending, then we should
```

Comment:
I don't know if this would make sense here, but would a user-provided timeout setting be useful if this is something that could exceed expectations?

Reply:
A user-provided timeout setting is not really useful in this case, as we would require the users to have a very thorough understanding of the internals of the client library for them to be able to set a sensible value. But a system default timeout value does make sense for this. A timeout should however not cause an error, but cause the pending requests to be ignored and then continue to create the database.